criu/pages-compress.c

```diff
@@ -0,0 +1,208 @@
+#include <limits.h>
+#include <pthread.h>
+#include <sched.h>
+#include <stdio.h>
+#include <sys/resource.h>
+#include <unistd.h>
+
+#include <compel/plugins/std/syscall-codes.h>
+
+#include "pages-compress.h"
+
+#include "crtools.h"
+#include "image.h"
+#include "log.h"
+#include "memfd.h"
+#include "sched.h"
+#include "util.h"
+
+#include <lz4/programs/lz4io.h>
+#include <lz4/lib/lz4frame.h>
+
+#undef LOG_PREFIX
+#define LOG_PREFIX "decompress: "
+
+const char *temp_path_decompressed = "/tmp/decompressed";
```

Comment:
The name should contain `getpid()` to prevent races. I would put there also `geteuid()` to prevent a failure with stale file.
A nitpick is that the code does not use the pointer itself. So better is to make the pointer also cost:
```
const char *const temp_path_decompressed = "...";
```
And even better is to remove the pointer:
```
const char temp_path_decompressed[] = "...";
```